tool.c

```diff
@@ -221,9 +222,105 @@ compdb(int argc, char *argv[])
 	return 0;
 }
 
+static void
+targets_rule(int argc, char *argv[])
+{
+	struct edge *e = NULL;
+	for (e = alledges; e; e = e->allnext) {
+		if (argc == 2) {
+			if (e->nin == 0)
+				continue;
+
+			if (!e->in[0]->gen)
+				printf("%s\n", e->in[0]->path->s);
+		}
+		else if (argc == 3) {
+			if (e->nout == 0)
+				continue;
+
+			if (strcmp(e->rule->name, argv[2]) == 0)
+				printf("%s\n", e->out[0]->path->s);
+		}
+	}
```

Comment:
I think you need to iterate through all inputs/outputs of the edge, not just the first one.

Reply:
Done